Sources/AblyAssetTrackingUI/LocationAnimator/DefaultLocationAnimatorCalculator+SupportingTypes.swift

```diff
@@ -0,0 +1,358 @@
+                case noLocations
+                /// There is a single location awaiting animation.
+                case singleLocation(Location)
+                /// There are at least two locations awaiting animation, in which case we also store information about how much of the journey between the first and second location we’ve already animated.
+                case multipleLocations(MultipleLocations)
+
+                /// Converts a `OneOrMoreLocationsAwaitingAnimation` value into a `LocationsAwaitingAnimation` value.
+                init(oneOrMoreLocationsAwaitingAnimation: MultipleLocations.OneOrMoreLocationsAwaitingAnimation) {
+                    switch oneOrMoreLocationsAwaitingAnimation {
+                    case let .singleLocation(location):
+                        self = .singleLocation(location)
+                    case let .multipleLocations(multipleLocations):
+                        self = .multipleLocations(multipleLocations)
+                    }
+                }
+
+                /// Contains information about two or more locations awaiting animation, as well as information about how much of the journey between the first and second location we’ve already animated.
+                struct MultipleLocations {
+                    /// The first location awaiting animation.
+                    var first: Location
+                    /// The second location awaiting animation.
+                    var second: Location
+                    /// The proportion of the journey between the first and second location we’ve already animated. Lies in (0...1).
+                    var proportionOfFirstToSecondAlreadyAnimated: Double {
+                        willSet {
+                            precondition((0 ... 1.0).contains(newValue))
+                        }
+                    }
+
+                    /// The remaining locations to be animated through, after `first` and `second`.
+                    var remaining: [Location]
+
+                    init(
+                        first: Location,
+                        second: Location,
+                        proportionOfFirstToSecondAlreadyAnimated: Double,
+                        remaining: [Location]
+                    ) {
+                        precondition((0 ... 1.0).contains(proportionOfFirstToSecondAlreadyAnimated))
+
+                        self.first = first
+                        self.second = second
+                        self.proportionOfFirstToSecondAlreadyAnimated = proportionOfFirstToSecondAlreadyAnimated
+                        self.remaining = remaining
+                    }
+
+                    /// Like `LocationsAwaitingAnimation`, contains information about the locations of the asset that we still need to animate through. It differs from `LocationsAwaitingAnimation` in that it always contains at least one location; that is, it has no `noLocations` case.
+                    enum OneOrMoreLocationsAwaitingAnimation {
```

Comment:
Shouldn't it be on the same level as `LocationsAwaitingAnimation` (and not embed into it)?

Reply:
I thought it made sense to put it into the narrowest scope it needs to be inside. Given that it's only used as a return value of `MultipleLocations.progressing(byNumberOfLocationTransitions:)`, that means inside `MultipleLocations`.